skip.js

```diff
@@ -1,18 +1,70 @@
+// Create an observer instance linked to the callback function
+let observer = null;
+
 function skip(button, inner_text) {
     let buttons = document.querySelectorAll(button);
 
     // Confirm we aren't about to click on something we don't intend to.
     for(let i = 0;i < buttons.length; ++i) {
-        if(buttons[i].innerText === inner_text)
+        if (buttons[i].innerText === inner_text) {
+            if (observer != null) {
+                observer.disconnect(); // with mission accomplished, deactivate observer
+                observer = null;
+            }
             buttons[i].click();
+        }
     }
 }
 
-switch (window.location.hostname) {
+function go() {
+    switch (window.location.hostname) {
     case 'delivery.cardago.com':
         skip('.btn_base', 'READY TO OPEN YOUR GIFT CARD?');
         break;
     case 'www.paypal.com':
         skip('[data-pa-click]', 'Skip');
         break;
+    }
+}
+
+function configure_dom_observer(id_to_observe) {
+    // Select the node that will be observed for mutations
+    let targetNode = document.getElementById(id_to_observe);
+
+    // Options for the observer (which mutations to observe)
+    const config = { attributes: true, childList: true, subtree: true };
+
+    // Callback function to execute when mutations are observed
+    const callback = function(mutationsList, observer) {
+        // Use traditional 'for loops' for IE 11
```

Comment:
Probably wouldn't need to support IE 11 from a Chromium/Firefox extension, but I was using the same MDN page as reference :)

Reply:
Yeah this was copy pasta from that MDN page... You can disregard that or even take away attributes from the config and remove that else if clause... I figure it doesn't hurt to have in case more sites go this route in the future - the plumbing is already there albeit commented out.